### PR TITLE
DropSection::CustomSectionByName to take a string by value

### PR DESCRIPTION
**Pull request by jakelang**

### CODE REVIEW

File: libchisel/src/dropsection.rs [hunk 1]
```diff
@@ -3,17 +3,17 @@ use super::{ChiselModule, ModuleError, ModuleKind, ModuleTranslator};
 use parity_wasm::elements::*;
 
 /// Enum on which ModuleTranslator is implemented.
-pub enum DropSection<'a> {
+pub enum DropSection {
     NamesSection,
     /// Name of the custom section.
-    CustomSectionByName(&'a String),
+    CustomSectionByName(String),
```
Comment (axic) [inline libchisel/src/dropsection.rs]: Don't know why this done this way.

**Comment by axic** [comment]
Looks good, but please first fix the CI.

File: libchisel/src/dropsection.rs (hunk 1 above)
Comment (jakelang) [reply libchisel/src/dropsection.rs]: A helper to construct DropSection which generates a string cannot return DropSection to caller if it contains a reference to a string whose scope is within the helper function.
There are workarounds but they aren't very ergonomic.

Comment (axic) [reply libchisel/src/dropsection.rs]: Fixed my typo: Don't know why it was done with a reference initially.

**Pull request merged**